File: backend/scratch/test_prod_api_call.py
import sys
import os

# Add the parent directory of the current directory to sys.path
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from app import create_app
from models import User
from flask_jwt_extended import create_access_token
import requests

def test_prod_api():
    app = create_app()
    with app.app_context():
        try:
            print("Fetching Prasanna Karagar user...")
            u = User.objects(email="ugcet2502059@reva.edu.in").first()
            if not u:
                print("Prasanna Karagar not found in DB!")
                return
            
            print(f"User role: {u.role}")
            
            # Let's generate a token
            token = create_access_token(identity=str(u.id))
            print(f"Generated JWT Token: {token[:20]}...")
            
            # Let's make an actual HTTP request to the production backend!
            url = "https://unirank-1.onrender.com/api/admin/users"
            headers = {
                "Authorization": f"Bearer {token}"
            }
            print(f"Sending GET request to {url}...")
            res = requests.get(url, headers=headers)
            print(f"Response status code: {res.status_code}")
            print(f"Response json: {res.json()}")
            
        except Exception as e:
            import traceback
            print(f"ERROR: {e}")
            traceback.print_exc()

if __name__ == "__main__":
    test_prod_api()

File: backend/src/scripts/scratch/testProdApiCall.js
/**
 * scripts/scratch/testProdApiCall.js
 */

import dotenv from 'dotenv';
import connectDB from '../../config/db.js';
import { User } from '../../models/index.js';
import { createAccessToken } from '../../middleware/auth.js';
import axios from 'axios';
import mongoose from 'mongoose';

dotenv.config();

async function main() {
  await connectDB();
  try {
    console.log('Fetching Prasanna Karagar user...');
    const u = await User.findOne({ email: 'ugcet2502059@reva.edu.in' });
    if (!u) {
      console.log('Prasanna Karagar not found in DB!');
      await mongoose.disconnect();
      return;
    }

    console.log(`User role: ${u.role}`);
    const token = createAccessToken(u._id.toString());
    console.log(`Generated JWT Token: ${token.substring(0, 20)}...`);

    const url = 'https://unirank-1.onrender.com/api/admin/users';
    console.log(`Sending GET request to ${url}...`);
    const res = await axios.get(url, { headers: { Authorization: `Bearer ${token}` } });
    console.log(`Response status code: ${res.status}`);
    console.log('Response json:', res.data);
  } catch (err) {
    console.error('ERROR:', err.message);
  }
  await mongoose.disconnect();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
